### PR TITLE
fix: Two way data binding

### PR DESCRIPTION
**Pull request by danharrin**
> Note: To avoid merge conflicts, this PR includes changes from both #1 and #2, so merge those first :)

This PR allows the editor to respond to state changes made from the backend as well as the frontend

Also, it enables state binding modifiers like `reactive()` that we have built-in to all Filament fields

### CODE REVIEW

File: resources/views/markdownField.blade.php [hunk 1]
```diff
@@ -20,124 +20,77 @@ function debounce(func, timeout = 300) {
                 }, timeout);
             };
         }
+    </script>
 
-        window.initMarkdownEditor = function () {
-            let editor = new EasyMDE({
+    <div
+        x-data="{ state: $wire.{{ $applyStateBindingModifiers('entangle(\'' . $getStatePath() . '\')') }} }"
```
Comment (danharrin) [inline resources/views/markdownField.blade.php:26]: `$applyStateBindingModifiers()` dynamically adds `.defer` to the field when it's not `reactive()`

`reactive()` means that the Livewire component will reload each time the editor changes

File: resources/views/markdownField.blade.php [hunk 2]
```diff
@@ -20,124 +20,77 @@ function debounce(func, timeout = 300) {
...
-        }
-    </script>
 
-    <div
-        x-data="{ state: $wire.entangle('{{ $getStatePath() }}').defer }"
-        x-init="
-            editor = new EasyMDE({ element: $refs.textarea })
-            editor.value(state ?? '')
-            editor.codemirror.on('change', () => state = editor.value())
             $watch('state', () => {
```
Comment (danharrin) [inline resources/views/markdownField.blade.php:84]: This watcher will be triggered when the state changes from the backend, for example when `$set()` is called

File: resources/views/markdownField.blade.php [hunk 3]
```diff
@@ -20,124 +20,77 @@ function debounce(func, timeout = 300) {
...
                 if (editor.codemirror.hasFocus()) {
-                    return
+                    return;
                 }
-                editor.value(state ?? '')
-            })
+
+                editor.value(state ?? '');
```
Comment (danharrin) [inline resources/views/markdownField.blade.php:89]: This sets the value of the editor to the new state

**Comment by freekmurze** [comment]
Thank you 🙌

**Pull request merged**